src/main/java/gil/server/HTTPParser.java

```diff
@@ -75,16 +89,61 @@ public String getRequestURI(String startLine) {
         return headersHash;
     }
 
+    public HashMap<String, String> getParameters(String requestLine) throws UnsupportedEncodingException {
+        HashMap<String, String> parametersHash = new HashMap<>();
+        String requestLineSeparator = " ";
+        int indexOfUriInRequestLine = 1;
```

Comment:
To echo the comment Cyrus left, I'd put this into a constant along with the other regex patterns since they are used more than once in this file.